src/main/scala/firrtl/annotations/Annotation.scala

```diff
@@ -28,6 +28,37 @@ trait NoTargetAnnotation extends Annotation {
   def update(renames: RenameMap) = Seq(this)
 }
 
+/** Indicates that this annotation ''cannot'' be serialized to JSON
+  *
+  * This is intended to be used if you have an annotation containing something that json4s cannot handle, e.g., an
+  * anonymous function like `() => Module`. You can optionally override [[toJsonSerializable]] to provide a conversion
+  * of this unserializable annotation to a serializable one. For example, a dut generator like `() => Module` can be
+  * serializable to a circuit via elaboration.
+  *
+  * '''Use this sparingly! If you find yourself using this, you are likely using an anti-pattern that can be better
+  * accomplished with a different approach!'''
+  */
+trait Unserializable extends Annotation {
+  /** A stub method that will be used to serialize this unserializable annotation to another [[Annotation]] that ''can''
+    * be serialized to JSON.
+    *
+    * '''This method must return an annotation that does not subclass [[Unserializable]]!'''
+    *
+    * @note This is not type safe. There may be a better way to do this...
+    * @throw AnnotationException if this method is implemented to return an [[Annotation]] that subclasses this trait
+    */
+  def toJsonSerializable: Annotation
+
+  /** Do the conversion to an [[Annotation]] that can be serialized to JSON. This will check that the conversion does not
+    * have the [[Unserializable]] trait.
+    */
+  final def _toJsonSerializable: Annotation = this.toJsonSerializable match {
+    case _: Unserializable =>
+      throw new AnnotationException(s"Annotation '${this.getClass.getName}' cannot be serialized to JSON")
+    case a => a
+  }
+}
+
```

Comment:
@chick: Here's what I came up with for using `Annotation`s for internal tool data movement of things which we can't serialize with json4s (currently). The idea here is that you mix this into an `Annotation` and before this is serialized to JSON, it will be converted to an annotation that _is serializable_. This is damn ugly as I wasn't able to think of a way to do this with type checking. Maybe there's a way with proper variances, but nothing was jumping out at me. This _should enable_ something (which I'll add to Chisel) of a `ChiselCircuitGeneratorAnnotation(dut: () => RawModule)` that will serialize to a `ChiselCircuitAnnotation(circuit: Circuit)` but can be used for Chisel testers.

@jackkoenig: This is to address a narrow problem of passing annotations into the `Driver` of Chisel testers. It needs a `() => MultiIOModule` to pass to a `PeekPokeTester` and this is information that can't be contained in a serializable annotation without completely rewriting how Chisel testers works. Can you think of any way to add type safety here?